### PR TITLE
feat: --env for cli

### PR DESCRIPTION
**Pull request by aarnphm**
- perf: improve import cli speed
- chore: move env to --

There are some typing refactoring here as well and we improve the runtime by 2ms :)

### CODE REVIEW

File: bentoml/_internal/cli/__init__.py [hunk 1]
```diff
@@ -1,15 +1,78 @@
...
+        if not value or ctx.resilient_parsing:
+            return
+
+        is_windows = sys.platform == "win32"
+
+        info_dict = {
+            "BentoML version": BENTOML_VERSION,
+            "Python version": platform.python_version(),
+            "Platform info": platform.platform(),
+            "Conda info": "not installed",
```
[the remark this answers is not in the document]
Comment (aarnphm) [reply bentoml/_internal/cli/__init__.py:48]: imo having uppercase is more presentable.

[the remark this answers is not in the document]
Comment (aarnphm) [reply bentoml/_internal/cli/__init__.py:48]: Screenshot 2022-07-08 at 11 16 56

**Comment by ssheng** [comment]
I ran into an issue with `conda`. This is may be a problem
Screen Shot 2022-07-08 at 16 45 51

File: bentoml/_internal/cli/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ssheng) [reply bentoml/_internal/cli/__init__.py]: I was thinking something simple like below. The not left-aligned formatting may be problematic in a non-monospaced text display.
```
bentoml: 1.0.0rc3.post17+g60c030f5
python: 3.9.12
platform: macOS-10.16-x86_64-i386-64bit
```

[the remark this answers is not in the document]
Comment (aarnphm) [reply bentoml/_internal/cli/__init__.py]: sure we can do something like this. I don't have a strong opinion either.

**Review by ssheng — APPROVED** [review]

**Pull request merged**